src/nodes/typeGuards/single.ts

```diff
@@ -0,0 +1,346 @@
+	isCallSignatureDeclaration,
+	isCaseBlock,
+	isCaseClause,
+	isCatchClause,
+	isClassDeclaration,
+	isClassElement,
+	isClassExpression,
+	isClassLike,
+	isClassOrTypeElement,
+	isClassStaticBlockDeclaration,
+	isCommaListExpression,
+	isComputedPropertyName,
+	isConditionalExpression,
+	isConditionalTypeNode,
+	isConstructorDeclaration,
+	isConstructorTypeNode,
+	isConstructSignatureDeclaration,
+	isContinueStatement,
+	isDebuggerStatement,
+	isDecorator,
+	isDefaultClause,
+	isDeleteExpression,
+	isDoStatement,
+	isDotDotDotToken,
+	isElementAccessExpression,
+	isEmptyStatement,
+	isEnumDeclaration,
+	isEnumMember,
+	isExportAssignment,
+	isExportDeclaration,
+	isExportSpecifier,
+	isExpressionStatement,
+	isExpressionWithTypeArguments,
+	isExternalModuleReference,
+	isForInStatement,
+	isForOfStatement,
+	isForStatement,
+	isFunctionDeclaration,
+	isFunctionExpression,
+	isFunctionTypeNode,
+	isGetAccessor,
+	isGetAccessorDeclaration,
+	isHeritageClause,
+	isIdentifier,
+	isIfStatement,
+	isImportClause,
+	isImportDeclaration,
+	isImportEqualsDeclaration,
+	isImportSpecifier,
+	isImportTypeAssertionContainer,
+	isImportTypeNode,
+	isIndexedAccessTypeNode,
+	isIndexSignatureDeclaration,
+	isInferTypeNode,
+	isInterfaceDeclaration,
+	isIntersectionTypeNode,
+	isJSDoc,
+	isJSDocAllType,
+	isJSDocAugmentsTag,
+	isJSDocAuthorTag,
+	isJSDocCallbackTag,
+	isJSDocClassTag,
+	isJSDocDeprecatedTag,
+	isJSDocEnumTag,
+	isJSDocFunctionType,
+	isJSDocImplementsTag,
+	isJSDocLink,
+	isJSDocLinkCode,
+	isJSDocLinkLike,
+	isJSDocLinkPlain,
+	isJSDocMemberName,
+	isJSDocNamepathType,
+	isJSDocNameReference,
+	isJSDocNonNullableType,
+	isJSDocNullableType,
+	isJSDocOptionalType,
+	isJSDocOverrideTag,
+	isJSDocParameterTag,
+	isJSDocPrivateTag,
+	isJSDocPropertyLikeTag,
+	isJSDocPropertyTag,
+	isJSDocProtectedTag,
+	isJSDocPublicTag,
+	isJSDocReadonlyTag,
+	isJSDocReturnTag,
+	isJSDocSeeTag,
+	isJSDocSignature,
+	isJSDocTemplateTag,
+	isJSDocThisTag,
+	isJSDocTypedefTag,
+	isJSDocTypeExpression,
+	isJSDocTypeLiteral,
+	isJSDocTypeTag,
+	isJSDocUnknownTag,
+	isJSDocUnknownType,
+	isJSDocVariadicType,
+	isJsxAttribute,
+	isJsxAttributes,
+	isJsxClosingElement,
+	isJsxClosingFragment,
+	isJsxElement,
+	isJsxExpression,
+	isJsxFragment,
+	isJsxOpeningElement,
+	isJsxOpeningFragment,
+	isJsxSelfClosingElement,
+	isJsxSpreadAttribute,
+	isJsxText,
+	isLabeledStatement,
+	isLiteralTypeNode,
+	isMappedTypeNode,
+	isMetaProperty,
+	isMethodDeclaration,
+	isMethodSignature,
+	isMinusToken,
+	isMissingDeclaration,
+	isModuleBlock,
+	isModuleDeclaration,
+	isNamedExports,
+	isNamedImports,
+	isNamedTupleMember,
+	isNamespaceExport,
+	isNamespaceExportDeclaration,
+	isNamespaceImport,
+	isNewExpression,
+	isNonNullExpression,
+	isNoSubstitutionTemplateLiteral,
+	isNotEmittedStatement,
+	isNumericLiteral,
+	isObjectBindingPattern,
+	isObjectLiteralElement,
+	isObjectLiteralExpression,
+	isOmittedExpression,
+	isOptionalTypeNode,
+	isParameter as isParameterDeclaration,
```

Comment:
Should we use `isParameter` or `isParameterDeclaration`?

The type is called `ParameterDeclaration` so I vote for `isParameterDeclaration` as opposed to TypeScript's name for this function.

Reply:
Agreed, TypeScript should switch to `isParameterDeclaration` 😄

Reply:
If we are not re-exporting TypeScript's type guards, should we just ignore this?

Reply:
Yeah makes sense